lecture_5/homework/app/services/reports/siege_report.rb

```diff
@@ -3,10 +3,23 @@
 module Reports
   class SiegeReport
     def initialize(building:)
+      @building = building
     end
 
     def call
-      raise NotImprementedYet
+      daily_supplies = 10
+      if @building.warriors.alive.any?
```

Comment:
To sprawdzenie spokojnie by się mogło znaleźć gdzieś wyżej, żeby nie obciążać logiki serwisu :) Serwis najlepiej jeśli ma jedną odpowiedzialność, a Twój serwis poza liczeniem ile dni przetrwa dany budynek, stwierdza upadłość budynku jeśli nie ma w nim wojowników :)